File: chapter5/php/com/dotComIt/learnWith/config/DatabaseConfig.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: jhouser
 * Date: 12/6/2017
 */

class DatabaseConfig
{
    // specify your own database credentials
    private $host = "developer.dot-com-it.com";
    private $port = "2433";
    private $db_name = "LearnWithApp";
    private $username = "LearnWithUser";
    private $password = "password";

    public $conn;

    // get the database connection
    public function openConnection(){

        $this->conn = null;

        // for non PDO Connection:
        $serverName = "tcp:" . $this->host . "," . $this->port;
        $connectionOptions = array("Database"=>$this->db_name,"Uid"=>$this->username, "PWD"=>$this->password);

        // for PDO Connection:
//        $serverName = "tcp:$this->host,$this->port";
        $serverName = "tcp:$this->host,$this->port";
        $connectionString = "sqlsrv:server=$serverName; Database=$this->db_name";

        try{
//            $this->conn = sqlsrv_connect($serverName, $connectionOptions);
            $this->conn = new PDO( $connectionString, $this->username, $this->password);

            if($this->conn == false){
                die(FormatErrors(sqlsrv_errors()));
            }
        }catch(Exception $exception){
            echo "Connection error: " . $exception->getMessage();
        }

        return $this->conn;
    }

    // close the database connection
    public function closeConnection(){
        sqlsrv_close($this->conn);
    }


}
?>

File: chapter1/java/src/main/java/com/dotComIt/learnWith/config/DatabaseConfig.java
package com.dotComIt.learnWith.config;

import java.sql.Connection;
import java.sql.DriverManager;
import com.microsoft.sqlserver.jdbc.SQLServerDriver;

public class DatabaseConfig {

    String host = "jdbc:sqlserver://developer.dot-com-it.com";
    // 1433 is the default SQL Server port
    String port = "2433";
    String database = "LearnWithApp";
    String username = "LearnWithUser";
    String password = "password";

    Connection connection = null;

    public Connection openConnection(){
        if(connection != null){
            return connection;
        }

        String connectionString = host + ":" + port + ";"
                + "database=" + database + ";"
                + "user=" + username + ";"
                + "password=" + password + ";";

        try {
            Class.forName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
            connection = DriverManager.getConnection(connectionString);
        }
        catch (Exception e) {
            System.out.println("Exception");
            e.printStackTrace();
        }
        finally {
            closeConnection();
        }
        return connection;

    };

    public void closeConnection(){
        try { connection.close(); } catch(Exception e) {}
    }

}
